src/services/services.ts

```diff
@@ -2504,14 +2542,26 @@ namespace ts {
             uncommentSelection,
         };
 
-        if (syntaxOnly) {
-            invalidOperationsOnSyntaxOnly.forEach(key =>
-                ls[key] = () => {
-                    throw new Error(`LanguageService Operation: ${key} not allowed on syntaxServer`);
-                }
-            );
+        switch (languageServiceMode) {
+            case LanguageServiceMode.Semantic:
+                break;
+            case LanguageServiceMode.ApproximateSemanticOnly:
+                invalidOperationsOnApproximateSemanticOnly.forEach(key =>
+                    ls[key] = () => {
+                        throw new Error(`LanguageService Operation: ${key} not allowed on approximate semantic only server`);

```

Comment:
I think it might be more helpful to say something like `not allowed in LanguageServiceMode.ApproximateSemanticOnly`, so that it's explicitly connected to the option.